cpp/src/arrow/compute/api_scalar.h

```diff
@@ -49,10 +49,52 @@ class ARROW_EXPORT ElementWiseAggregateOptions : public FunctionOptions {
   explicit ElementWiseAggregateOptions(bool skip_nulls = true);
   constexpr static char const kTypeName[] = "ElementWiseAggregateOptions";
   static ElementWiseAggregateOptions Defaults() { return ElementWiseAggregateOptions{}; }
-
   bool skip_nulls;
 };
 
+/// Rounding modes for Round and MRound functions. General modes are prefixed
+/// with TOWARDS and tie-breaker modes are prefixed with HALF. Common aliases
+/// are available for several modes.
```

Comment:
Are the aliases worth keeping? Maybe it's better to just document the various names for them but keep only a single name for each possible case.

Reply:
IMHO, aliases are not worth keeping, but I was not sure what API to use:
* aliases are based on [C++ rounding modes](https://en.cppreference.com/w/cpp/numeric/fenv/FE_round) (which is a subset)
* "true" rounding modes are based on [general definition](https://en.wikipedia.org/wiki/Rounding#Rounding_to_integer)

Reply:
Related to aliases, there are [rounding modes that are equivalent to other compute functions](https://github.com/apache/arrow/pull/10349/files#diff-3eafd7246f6a8c699f10d46e3276852fe44b6853b5517ef10396e561730c09f4R874-R887):
* `TOWARDS_NEG_INFINITY` -> `Floor`
* `TOWARDS_POS_INFINITY` -> `Ceil`
* `TOWARDS_ZERO` -> `Trunc`

Do you recommend for `Round/MRound` to invoke `Floor/Ceil/Trunc` or invoke `std::floor/ceil/trunc`? Currently, there is a bit of code duplication in the function bodies as both use the `std` variants but this to bypass unneeded overhead.

Reply:
So long as they are consistent it should be ok. It sounds like the general definition would be best.

I think the duplication is ok. If there's no difference we might actually prefer to have Floor/Ceil/Trunc be aliases or call into the more general kernel.